packages/jellyfish-api-core/__tests__/category/loan/setLoanToken.test.ts

```diff
@@ -0,0 +1,258 @@
+    const index = Object.keys(data).indexOf(loanTokenId) + 1
+    expect(data[loanTokenId].token[index].name).toStrictEqual('x'.repeat(128)) // Only remain the first 128 letters.
+  })
+
+  it('should not setLoanToken if priceFeedId does not contain USD price', async () => {
+    const priceFeedId: string = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token6',
+      currency: 'Token6'
+    }], 1])
+    await testing.generate(1)
+
+    const promise = testing.rpc.loan.setLoanToken({
+      symbol: 'Token6',
+      name: 'Token6',
+      priceFeedId
+    })
+    await expect(promise).rejects.toThrow(`RpcApiError: 'Test LoanSetLoanTokenTx execution failed:\noracle (${priceFeedId}) does not contain USD price for this token!', code: -32600, method: setloantoken`)
+  })
+
+  it('should not setLoanToken if priceFeedId is invalid', async () => {
+    const promise = testing.rpc.loan.setLoanToken({
+      symbol: 'Token7',
+      name: 'Token7',
+      priceFeedId: 'e40775f8bb396cd3d94429843453e66e68b1c7625d99b0b4c505ab004506697b'
+    })
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test LoanSetLoanTokenTx execution failed:\noracle (e40775f8bb396cd3d94429843453e66e68b1c7625d99b0b4c505ab004506697b) does not exist or not valid oracle!\', code: -32600, method: setloantoken')
+  })
+
+  it('should setLoanToken if mintable is false', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token8',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const loanTokenId = await testing.rpc.loan.setLoanToken({
+      symbol: 'Token8',
+      name: 'Token8',
+      priceFeedId,
+      mintable: false
+    })
+    expect(typeof loanTokenId).toStrictEqual('string')
+    expect(loanTokenId.length).toStrictEqual(64)
+    await testing.generate(1)
+  })
+
+  it('should setLoanToken if interest is greater than 0', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token9',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const loanTokenId = await testing.rpc.loan.setLoanToken({
+      symbol: 'Token9',
+      name: 'Token9',
+      priceFeedId,
+      interest: new BigNumber(0.2)
+    })
+    expect(typeof loanTokenId).toStrictEqual('string')
+    expect(loanTokenId.length).toStrictEqual(64)
+    await testing.generate(1)
+  })
+
+  it('should not setLoanToken if interest is less than 0', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token10',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const promise = testing.rpc.loan.setLoanToken({
+      symbol: 'Token10',
+      name: 'Token10',
+      priceFeedId,
+      interest: new BigNumber(-0.01)
```

Comment:
one more test for 
interest: new BigNumber(0.123456789)

Reply:
Rather than add a new test item, I change the value here

https://github.com/DeFiCh/jellyfish/pull/568/files#diff-d8ff95ade762b68edc7365bb17e82d9054f624c2402e865e51da88bba9059392R193

Reply:
I just discovered that they do not allow number that have more than 8 digit in the fractional part. I shall add a test unit for this